### ISSUE TITLE
CRAN Check Failure for Upcoming broom Release

### ISSUE THREAD

**Issue by simonpcouch** [stefvanbuuren/mice#237]
Hi there! The broom dev team just ran reverse dependency checks on the upcoming broom 0.7.0 release and found new errors/test failures for the CRAN version of this package. I've pasted the results below, which seem to result from adjustments to glance methods for `lm` and `aov` objects.

*   checking examples ... ERROR
    ```
    Running examples in ‘mice-Ex.R’ failed
    The error most likely occurred in:
    
    > ### Name: D3
    > ### Title: Compare two nested models using D3-statistic
    > ### Aliases: D3
    > 
    > ### ** Examples
    > 
    > # Compare two linear models:
    > imp <- mice(nhanes2, seed = 51009, print = FALSE)
    > mi1 <- with(data = imp, expr = lm(bmi ~ age + hyp + chl))
    > mi0 <- with(data = imp, expr = lm(bmi ~ age + hyp))
    > D3(mi1, mi0)
    Error in eval_tidy(xs[[j]], mask) : object 'fstatistic' not found
    Calls: D3 ... with.default -> eval -> eval -> tibble -> tibble_quos -> eval_tidy
    Execution halted
    ```

*   checking tests ...
    ```
     ERROR
    Running the tests in ‘tests/testthat.R’ failed.
    Last 13 lines of output:
      [90m 19. [39mbroom:::glance.lm(X[[i]], ...)
      [90m 21. [39mbase::with.default(...)
      [90m 22. [39m[ base::eval(...) ][90m with 1 more call[39m
      [90m 24. [39mtibble::tibble(...)
      [90m 25. [39mtibble:::tibble_quos(xs[!is_null], .rows, .name_repair)
      [90m 26. [39mrlang::eval_tidy(xs[[j]], mask)
      
      ══ testthat results  ═══════════════════════════════════════════════════════════
      [ OK: 280 | SKIPPED: 0 | WARNINGS: 1 | FAILED: 3 ]
      1. Error: (unknown) (@test-D3.R#10) 
      2. Error: anova.mira() produces silent D1 and D3 (@test-anova.R#9) 
      3. Error: anova.mira() produces warnings on D2 (@test-anova.R#14) 
      
      Error: testthat unit tests failed
      Execution halted
    ```

Two notes from `NEWS.md` that might be relevant here:

> We have changed how we report degrees of freedom for `lm` objects (tidymodels/broom#212, tidymodels/broom#273). This is especially important for instructors in statistics courses. Previously the `df` column in `glance.lm()` reported the rank of the design matrix. Now it reports degrees of freedom of the numerator for the overall F-statistic. This is equal to the rank of the model matrix minus one (unless you omit an intercept column), so the new `df` should be the old `df` minus one.

> We have simplified `glance.aov()`, which now contains only the following columns: `logLik`, `AIC`, `BIC, deviance`, `df.residual`, `nobs` (see tidymodels/broom#212). Note that `tidy.aov()` gives more complete information about degrees of freedom in an `aov` object.

We hope to submit this new version of the package to CRAN in the coming weeks. If you encounter any problems fixing these issues, please feel free to reach out!🙂

**Comment by stefvanbuuren** [comment]
There are 3 error errors that need to be fixed.

With `broom 0.7.0.9000` I get:

```r
> checking tests ...
  See below...

── Test failures ───────────────────────────────────────────────── testthat ────

> library(testthat)
> library(mice)

Attaching package: 'mice'

The following objects are masked from 'package:base':

    cbind, rbind

> 
> test_check("mice")
── 1. Error: (unknown) (@test-D3.R#10)  ──────────────────────────
object 'fstatistic' not found
Backtrace:
  1. mice::D3(fit1 = fit1, fit0 = fit0)
  5. base::lapply(fit0, glance)
  7. broom:::glance.lm(X[[i]], ...)
  9. base::with.default(...)
 10. [ base::eval(...) ] with 1 more call
 12. tibble::tibble(...)
 13. tibble:::tibble_quos(xs[!is_null], .rows, .name_repair)
 14. rlang::eval_tidy(xs[[j]], mask)

── 2. Error: anova.mira() produces silent D1 and D3 (@test-anova.R
object 'fstatistic' not found
Backtrace:
  1. testthat::expect_silent(z1 <- anova(m2, m1, m0))
 10. mice:::anova.mira(m2, m1, m0)
  3. base::lapply(modlist, getfit, 1L)
 11. base::sapply(., glance)
 20. base::lapply(X = X, FUN = FUN, ...)
 22. broom:::glance.lm(X[[i]], ...)
 24. base::with.default(...)
 25. [ base::eval(...) ] with 1 more call
 27. tibble::tibble(...)
 28. tibble:::tibble_quos(xs[!is_null], .rows, .name_repair)
 29. rlang::eval_tidy(xs[[j]], mask)

── 3. Error: anova.mira() produces warnings on D2 (@test-anova.R#1
object 'fstatistic' not found
Backtrace:
  1. testthat::expect_warning(z2a <- anova(m2, m1, m0, method = "D2"))
  7. mice:::anova.mira(m2, m1, m0, method = "D2")
  3. base::lapply(modlist, getfit, 1L)
 11. base::sapply(., glance)
 17. base::lapply(X = X, FUN = FUN, ...)
 19. broom:::glance.lm(X[[i]], ...)
 21. base::with.default(...)
 22. [ base::eval(...) ] with 1 more call
 24. tibble::tibble(...)
 25. tibble:::tibble_quos(xs[!is_null], .rows, .name_repair)
 26. rlang::eval_tidy(xs[[j]], mask)

══ testthat results  ═════════════════════════════════════════════
[ OK: 280 | SKIPPED: 0 | WARNINGS: 0 | FAILED: 3 ]
1. Error: (unknown) (@test-D3.R#10) 
2. Error: anova.mira() produces silent D1 and D3 (@test-anova.R#9) 
3. Error: anova.mira() produces warnings on D2 (@test-anova.R#14) 

Error: testthat unit tests failed
Execution halted

3 errors x | 0 warnings ✓ | 0 notes ✓
Error: R CMD check found ERRORs
Execution halted

Exited with status 1.
```

**Comment by stefvanbuuren** [comment]
I find that `broom 0.7.0.9000` throws the following error when the model contains an intercept only. Will open an issue at `modelverse/broom`.

```r
> library(broom)
> fit1 <- lm(mpg ~ disp, data = mtcars)
> glance(fit1)
# A tibble: 1 x 12
  r.squared adj.r.squared sigma statistic  p.value    df logLik   AIC   BIC deviance df.residual  nobs
      <dbl>         <dbl> <dbl>     <dbl>    <dbl> <dbl>  <dbl> <dbl> <dbl>    <dbl>       <int> <int>
1     0.718         0.709  3.25      76.5 9.38e-10     1  -82.1  170.  175.     317.          30    32
> fit0 <- lm(mpg ~ 1, data = mtcars)
> glance(fit0)
Error in eval_tidy(xs[[j]], mask) : object 'fstatistic' not found
> 
```

**Comment by stefvanbuuren** [comment]
After the [patch in broom](https://github.com/tidymodels/broom/commit/4600d9f2ff3aa314f45c10b9bafe99ba3418124b) the following problem remains:

```
> test_check("mice")
── 1. Error: (unknown) (@test-D3.R#58)  ──────────────────────────
No tidy method for objects of class lmerMod
Backtrace:
  1. mice::D3(fit1, fit0)
  2. mice::pool(fit1)
  3. mice:::pool.fitlist(getfit(object), dfcom = dfcom)
  5. mice:::summary.mira(fitlist, type = "tidy", exponentiate = FALSE)
  9. base::lapply(...)
 11. broom:::tidy.default(X[[i]], ...)

══ testthat results  ═════════════════════════════════════════════
[ OK: 286 | SKIPPED: 0 | WARNINGS: 0 | FAILED: 1 ]
1. Error: (unknown) (@test-D3.R#58) 

Error: testthat unit tests failed
Execution halted

1 error x | 0 warnings ✓ | 0 notes ✓
Error: R CMD check found ERRORs
Execution halted

Exited with status 1.
```

**Comment by stefvanbuuren** [comment]
Solved in 9184f07d2ad4dbce5de15453656c740dc2d51cdf.

Will go into the next CRAN release.
